### PR TITLE
polkadotjs article scripts fixes

### PR DESCRIPTION
**Pull request by augustocollado**
### Description

This PR fixes the scripts included in the polkadot js article

### Checklist

- [x] I have added a label to this PR 🏷️
- [x] I have run my changes through Grammarly
- [x] If this page requires a disclaimer, I have added one
- [x] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects

### CODE REVIEW

File: .snippets/code/builders/interact/substrate-api/polkadot-js-api/balance-event.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import '@polkadot/api-augment';
```
Comment (eshaben) [inline .snippets/code/builders/interact/substrate-api/polkadot-js-api/balance-event.ts:1]: I get tons of errors with this and tons without haha, so I don't know what's going on here - did you try this out?

File: builders/interact/substrate-api/polkadot-js-api.md [hunk 2]
```diff
@@ -352,8 +352,13 @@ import { encode } from '@polkadot/util-rlp';
 import { keccakAsHex } from '@polkadot/util-crypto';
 import { numberToHex } from '@polkadot/util';
 
+// Set the key type to string
+type txType = {
+    [key: string]: any
+}
```
Comment (eshaben) [inline builders/interact/substrate-api/polkadot-js-api.md:358]:
```suggestion
type txType = {
  [key: string]: any;
};
```

**Review by eshaben — COMMENTED** [review]
I think the Polkadot.js API page needs to be cleaned up a bit more. I didn't realize or maybe it just never registered fully that this page has JavaScript and TypeScript examples on it, but nowhere do we say anything about TypeScript; it does say in the intro to have Node.js installed, but never anything else. We also use `@polkadot/*` packages throughout this guide that we never instruct users to install.

File: .snippets/code/builders/interact/substrate-api/polkadot-js-api/balance-event.ts (hunk 1 above)
Comment (augustocollado) [reply .snippets/code/builders/interact/substrate-api/polkadot-js-api/balance-event.ts:1]: Hmmm, yes, I tested all the scripts copy/pasting from the article but probably I had something else installed that made it work.
I'll try again in using a fresh environment!

**Comment by augustocollado** [comment]
> I think the Polkadot.js API page needs to be cleaned up a bit more. I didn't realize or maybe it just never registered fully that this page has JavaScript and TypeScript examples on it, but nowhere do we say anything about TypeScript; it does say in the intro to have Node.js installed, but never anything else. We also use `@polkadot/*` packages throughout this guide that we never instruct users to install.

I'll go over this article again, in fact, the ticket I had created in jira said about using .ts extensions!

File: .snippets/code/builders/interact/substrate-api/polkadot-js-api/balance-event.ts (hunk 1 above)
Comment (eshaben) [reply .snippets/code/builders/interact/substrate-api/polkadot-js-api/balance-event.ts:1]: ok thanks!

**Comment by augustocollado** [comment]
I added some clarifications and the extra library that is required to run the examples.
Tested all the scripts starting from a fresh install, and now all the scripts run correctly :)

File: .snippets/text/_common/general-js-tutorial-check.md [hunk 3]
```diff
@@ -2,4 +2,6 @@
 !!! note
     The examples in this guide are based on a MacOS or Ubuntu 20.04 environment. If you're using Windows, you'll need to adapt them accordingly.
     
-    Furthermore, please ensure that you have Node.js and a package manager (such as npm or yarn) installed. Also, make sure you've initialized a  `package.json` file for ES6 modules. To learn how to install Node.js, please check their [official documentation](https://nodejs.org/en/download).
\ No newline at end of file
+    Furthermore, please ensure that you have Node.js and a package manager (such as npm or yarn) installed. To learn how to install Node.js, please check their [official documentation](https://nodejs.org/en/download).
```
Comment (eshaben) [inline .snippets/text/_common/general-js-tutorial-check.md:5]:
```suggestion
    Furthermore, please ensure that you have Node.js and a package manager (such as npm or yarn) installed. To learn how to install Node.js, please check their [official documentation](https://nodejs.org/en/download){target=\blank}. 
```

File: dapp-developers/developer-toolkit/substrate-api/polkadot-js-api.md [hunk 4]
```diff
@@ -11,31 +11,35 @@ description: Learn how to use the Polkadot.js API to interact with a Tanssi appc
...
-First, you need to install the Polkadot.js API library for your project through a package manager, such as `yarn`. Install it in your project directory with the following command:
+First, you need to install the Polkadot.js API library and the RLP library through a package manager such as `yarn`. Both dependencies are required to run the examples in this guide successfully.
+
+Install them in your project directory with the following command:
 
 === "npm"
 
     ```bash
     npm i @polkadot/api
+    npm i @polkadot/util-rlp
```
Comment (eshaben) [inline dapp-developers/developer-toolkit/substrate-api/polkadot-js-api.md:28]: Other libraries that are used that we should have them install now:

@polkadot/util
@polkadot/util-crypto
@polkadot/keyring

Maybe break it up and have @polkadot/api and then say depending on the example, you may need these additional packages 🤷‍♀️

**Review by eshaben — COMMENTED** [review]
Awesome!! For my own sanity, can you please update all of the code snippet files to `.ts` files and update all of the code block languages from `js` or `javascript` to `ts`?

- In the #sending-basic-transactions section, the `encodedCalldata` variable is then logged as `encodedCallData` - the capitalization needs to be fixed here

- In some of the transaction snippets, we use `12345` as the value to send. We should just update it to `INSERT_VALUE`, as we do with the addresses

- There should be an `await` added to this function for the batch transaction:
    ```ts
    // Construct the batch and send the transactions
    api.tx.utility
    ```
- The way we're using`at` has been deprecated. So, in the Sample Code for Monitoring Native Token Transfers snippet, the following change should be made from:
    ```ts
    polkadotApi.query.system.events.at(lastFinalizedHeader.hash)
    ```
    to:
    ```ts
    (await polkadotApi.at(lastFinalizedHeader.hash)).query.system.events()
    ```

File: dapp-developers/developer-toolkit/substrate-api/polkadot-js-api.md (hunk 4 above)
Comment (augustocollado) [reply dapp-developers/developer-toolkit/substrate-api/polkadot-js-api.md:28]: It seems that there is no need to explicitly install util, util-crypto, and keyring, those were installed along with "@polkadot/api"
(The only dependency that was missing when I tested the scripts was @polkadot/util-rlp).
Can you confirm?

**Comment by augustocollado** [comment]
Nice catches!!!

> Awesome!! For my own sanity, can you please update all of the code snippet files to `.ts` files and update all of the code block languages from `js` or `javascript` to `ts`? 😂
> 
> * In the #sending-basic-transactions section, the `encodedCalldata` variable is then logged as `encodedCallData` - the capitalization needs to be fixed here
> * In some of the transaction snippets, we use `12345` as the value to send. We should just update it to `INSERT_VALUE`, as we do with the addresses
> * There should be an `await` added to this function for the batch transaction:
>   ```ts
>   // Construct the batch and send the transactions
>   api.tx.utility
>   ```
> * The way we're using`at` has been deprecated. So, in the Sample Code for Monitoring Native Token Transfers snippet, the following change should be made from:
>   ```ts
>   polkadotApi.query.system.events.at(lastFinalizedHeader.hash)
>   ```
>   
>   
>       
>         
>       
>   
>         
>       
>   
>       
>     
>   to:
>   ```ts
>   (await polkadotApi.at(lastFinalizedHeader.hash)).query.system.events()
>   ```

File: dapp-developers/developer-toolkit/substrate-api/polkadot-js-api.md (hunk 4 above)
Comment (eshaben) [reply dapp-developers/developer-toolkit/substrate-api/polkadot-js-api.md:28]: you're correct! thanks for being on top of it!! 🙌

**Review by eshaben — APPROVED** [review]
LGTM! ✨

**Pull request merged**